.github/workflows/autoapprove.yml

```diff
@@ -0,0 +1,22 @@
+name: github-actions auto-approve
+on: pull_request_target
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  approve-bot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'github-actions' || github.actor == 'semgrep-ci[bot]'}}
```

Comment:
Is this a potential security issue?

Reply:
I think this is OK - what contexts will this be running in?

Reply:
It's mainly to auto-approve bump-version PRs in this repository

Reply:
do we *need* the `github-actions` actor? Can we restrict auto-approve and auto-merge to just the Semgrep CI bot?

The wrinkle here is I'm not sure this prevents somebody from e.g. forking this repo, adding a malicious workflow (or pre-commit step), then having their own GitHub action check out the evil fork and opening a PR. See https://github.com/orgs/community/discussions/49745#discussioncomment-5275999

Reply:
confirmed that this attack doesn't work on forks with our current auth model. A legitimate contributor *could* use this to skirt PR approval gates though

Reply:
Updated `autoapprove.yml` to use `github.event.pull_request.user.login`: thank you so much!